lib/matplotlib/tests/test_figure.py

```diff
@@ -496,6 +498,24 @@ def test_savefig_backend():
         fig.savefig("test.png", backend="pdf")
 
 
+@pytest.mark.parametrize('backend', ['Agg', 'Cairo'])
+def test_savefig_pixel_ratio(backend):
+    plt.switch_backend(backend)
+
+    fig = plt.figure()
+    with io.BytesIO() as buf:
+        fig.savefig(buf, format='png')
+        ratio1 = Image.open(buf)
+
+    fig = plt.figure()
+    fig.canvas._set_device_pixel_ratio(2)
+    with io.BytesIO() as buf:
+        fig.savefig(buf, format='png')
+        ratio2 = Image.open(buf)
+
+    assert ratio1.size == ratio2.size
```

Comment:
I guess even the whole images should be the same?

Reply:
True, but they're empty figures, so I didn't bother. I thought of using `check_figures_equal`, but I don't know how to force a different backend with it.

Reply:
You could actually plot something, and do a plain rgba array equality check?

Reply:
Sure